zebra-network/src/isolated/tests/vectors.rs

```diff
@@ -0,0 +1,220 @@
+    let mut outbound_join_handle = tokio::spawn(connect_isolated_tcp_direct(
+        network,
+        listen_addr,
+        "".to_string(),
+    ));
+
+    let (inbound_conn, _) = listener.accept().await.unwrap();
+
+    let mut inbound_stream =
+        Framed::new(inbound_conn, Codec::builder().for_network(network).finish());
+
+    // We don't need to send any bytes to get a version message.
+    if let Message::Version {
+        version,
+        services,
+        timestamp,
+        address_recv,
+        address_from,
+        nonce: _,
+        user_agent,
+        start_height,
+        relay,
+    } = inbound_stream
+        .next()
+        .await
+        .expect("stream item")
+        .expect("item is Ok(msg)")
+    {
+        // Check that the version message sent by connect_isolated
+        // anonymises all the fields that it possibly can.
+        //
+        // The version field needs to be accurate, because it controls protocol features.
+        // The nonce must be randomised for security.
+        //
+        // SECURITY TODO: check if the timestamp field can be zeroed, to remove another distinguisher (#3300)
+
+        let mut fixed_isolated_addr: SocketAddr = "0.0.0.0:0".parse().unwrap();
+        fixed_isolated_addr.set_port(network.default_port());
+
+        // Required fields should be accurate and match most other peers.
+        // (We can't test nonce randomness here.)
+        assert_eq!(version, CURRENT_NETWORK_PROTOCOL_VERSION);
+        assert_eq!(timestamp.timestamp() % (5 * 60), 0);
+
+        // Other fields should be empty or zeroed.
+        assert_eq!(services, PeerServices::empty());
+        assert_eq!(
+            address_recv,
+            // Since we're connecting to the peer, we expect it to have the node flag.
+            //
+            // SECURITY TODO: should this just be zeroed anyway? (#3300)
+            AddrInVersion::new(fixed_isolated_addr, PeerServices::NODE_NETWORK),
+        );
+        assert_eq!(
+            address_from,
+            AddrInVersion::new(fixed_isolated_addr, PeerServices::empty()),
+        );
+        assert_eq!(user_agent, "");
+        assert_eq!(start_height.0, 0);
+        assert!(!relay);
+    } else {
+        panic!("handshake did not send version message");
+    }
+
+    // Let the spawned task run for a short time.
+    tokio::time::sleep(Duration::from_secs(1)).await;
+
+    // Make sure that the isolated connection did not:
+    // - panic, or
+    // - return a service.
+    //
+    // This test doesn't send a version message on `inbound_conn`,
+    // so providing a service is incorrect behaviour.
+    //
+    // A timeout error would be acceptable,
+    // but a TCP connection error indicates a potential test setup issue.
+    // So we fail on them both, because we expect this test to complete before the timeout.
+    let outbound_result = futures::poll!(&mut outbound_join_handle);
+    assert!(matches!(outbound_result, Poll::Pending));
```

Comment:
Idea: this could be a timeout waiting for the task instead. Something like:

```suggestion
    // Make sure that the isolated connection did not:
    // - panic, or
    // - return a service.
    //
    // This test doesn't send a version message on `inbound_conn`,
    // so providing a service is incorrect behaviour.
    //
    // A timeout error would be acceptable,
    // but a TCP connection error indicates a potential test setup issue.
    // So we fail on them both, because we expect this test to complete before the timeout.
    let outbound_result =
        tokio::time::timeout(Duration::from_secs(1), &mut outbound_join_handle).await;
    assert!(outbound_result.is_err());
```

Reply:
I don't think there is any functional difference between the code, but I think the current code is slightly clearer.

In particular, in the suggested code, "timeout error" is ambiguous.